src/Facility.Core/Http/HttpClientService.cs

```diff
@@ -101,8 +103,66 @@ protected HttpClientService(HttpClientServiceSettings? settings, Uri? defaultBas
 			{
 				var contentType = mapping.RequestBodyContentType ?? requestHeaders?.GetContentType();
 				httpRequest.Content = GetHttpContentSerializer(requestBody.GetType()).CreateHttpContent(requestBody, contentType);
-				if (m_disableChunkedTransfer)
+				if (m_shouldCompressRequest(request))
+					httpRequest.Content = await CompressContentAsync(httpRequest.Content, cancellationToken).ConfigureAwait(false);
+				else if (m_disableChunkedTransfer)
 					await httpRequest.Content.LoadIntoBufferAsync().ConfigureAwait(false);
+
+				async static Task<HttpContent> CompressContentAsync(HttpContent httpContent, CancellationToken cancellationToken)
+				{
+					Stream? contentStream = null;
+					Stream? compressedContentStream = null;
+
+					try
+					{
+						// copy the existing HTTP content into a memory stream
+						contentStream = new MemoryStream();
```

Comment:
Worth using `RecyclableMemoryStream` or [supporting it somehow](https://github.com/FacilityApi/FacilityCSharp/blob/master/src/Facility.Core/Http/JsonHttpContentSerializerSettings.cs)?

Reply:
Not opposed. Add it in "V2" when we decide we need it?